src/function_space_operators.jl

```diff
@@ -0,0 +1,49 @@
+
+"""
+    GlaubitzNordströmÖffner2023()
+
+Function space SBP (FSBP) operators given in
+- Glaubitz, Nordström, Öffner (2023)
+  Summation-by-parts operators for general function spaces.
+  SIAM Journal on Numerical Analysis 61, 2, pp. 733-754.
+
+See also
+- Glaubitz, Nordström, Öffner (2024)
+  An optimization-based construction procedure for function space based
+  summation-by-parts operators on arbitrary grids.
+  arXiv, arXiv:2405.08770v1.
+"""
+struct GlaubitzNordströmÖffner2023 <: SourceOfCoefficients end
+
+function Base.show(io::IO, source::GlaubitzNordströmÖffner2023)
+  if get(io, :compact, false)
+    summary(io, source)
+  else
+      print(io,
+          "Glaubitz, Nordström, Öffner (2023) \n",
+          "  Summation-by-parts operators for general function spaces \n",
+          "  SIAM Journal on Numerical Analysis 61, 2, pp. 733-754. \n",
+          "See also \n",
+          "  Glaubitz, Nordström, Öffner (2024) \n",
+          "  An optimization-based construction procedure for function \n",
+          "    space based summation-by-parts operators on arbitrary grids \n",
+          "  arXiv, arXiv:2405.08770v1.")
+  end
+end
+
+"""
+    FunctionSpaceOperator(basis_functions, x_min, x_max, nodes, source; accuracy_order = 0)
+
+Construct a `MatrixDerivativeOperator` that represents a derivative operator in a function space.
+"""
+function FunctionSpaceOperator(basis_functions,
+                               x_min::T, x_max::T, nodes::Vector{T},
+                               source::SourceOfCoefficients;
+                               accuracy_order = 0) where {T, SourceOfCoefficients}
+
+  weights, D = construct_function_space_operator(basis_functions, x_min, x_max, nodes, source)
+  return MatrixDerivativeOperator(x_min, x_max, nodes, weights, D, accuracy_order, source)
+end
```

Comment:
We should document that users need to load `Optim` and `ForwardDiff` for this.

Reply:
I added a sentence to the docstring in [1b67472](https://github.com/ranocha/SummationByPartsOperators.jl/pull/268/commits/1b67472f663e8382bbf48a5d15f37dc3706c9c58). But actually I was wondering about this: Locally, it works fine for me without `using ForwardDiff`. `using SummationByPartsOperators, Optim` suffices for me to load the extension, which also confused me. Does that make sense or do you have an idea why I don't need to load ForwardDiff explicitly?

Reply:
Do you have it in your Project (or the default environment)?

Reply:
Yes, it was in my Project.toml (but it is not in my default environment), but I just tested that it also works without ForwardDiff being in my Project.toml. Also with ForwardDiff.jl in my Project `using SummationByPartsOperators, ForwardDiff` is not enough to call `function_space_operator`. I need `using Optim`, but not `using ForwardDiff`.

Reply:
Since Optim.jl depends on ForwardDiff.jl this makes sense. I updated the docstring accordingly.